homeassistant/components/lock/bmw_connected_drive.py

```diff
@@ -71,14 +71,16 @@ def is_locked(self):
     def lock(self, **kwargs):
         """Lock the car."""
         _LOGGER.debug("%s: locking doors", self._vehicle.modelName)
-        self._vehicle.remote_services.trigger_remote_door_lock()
         self._state = STATE_LOCKED
```

Comment:
Please add a comment before this, saying why we need a optimistic state update, when we also have a update callback.

Reply:
Added the comments